### PR TITLE
ES2015ify

### PR DESCRIPTION
**Pull request by nicolasbrugneaux**
Hey @sindresorhus , this pull-request addresses #90 , I'm not quite sure how to get rid of the lazy requires, so I didn't :sweat_smile:

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,102 +1,108 @@
...
+const lazyReq = require('lazy-req')(require);
+const lodash = lazyReq('lodash');
+const jshint = lazyReq('jshint');
+const jsxhint = lazyReq('jshint-jsx');
+const cli = lazyReq('jshint/src/cli');
+const loadConfig = lazyReq('./load-config');
+let plugin = {};
+let _;
+let markersByEditorId = {};
+let errorsByEditorId = {};
```
Comment (sindresorhus) [inline index.js]: Could be `const`

File: index.js [hunk 2]
```diff
@@ -1,102 +1,108 @@
...
+
+const lazyReq = require('lazy-req')(require);
+const lodash = lazyReq('lodash');
+const jshint = lazyReq('jshint');
+const jsxhint = lazyReq('jshint-jsx');
+const cli = lazyReq('jshint/src/cli');
+const loadConfig = lazyReq('./load-config');
+let plugin = {};
+let _;
+let markersByEditorId = {};
```
Comment (sindresorhus) [inline index.js]: Could be `const`

File: index.js [hunk 3]
```diff
@@ -1,102 +1,108 @@
...
+import path from 'path';
+import reactDomPragma from 'react-dom-pragma';
+
+const lazyReq = require('lazy-req')(require);
+const lodash = lazyReq('lodash');
+const jshint = lazyReq('jshint');
+const jsxhint = lazyReq('jshint-jsx');
+const cli = lazyReq('jshint/src/cli');
+const loadConfig = lazyReq('./load-config');
+let plugin = {};
```
Comment (sindresorhus) [inline index.js]: Could be `const`

File: index.js [hunk 4]
```diff
@@ -1,102 +1,108 @@
...
-var _;
-var markersByEditorId = {};
-var errorsByEditorId = {};
-var subscriptionTooltips = new CompositeDisposable();
+import { CompositeDisposable } from 'atom';
+import emissary from 'emissary';
+import path from 'path';
+import reactDomPragma from 'react-dom-pragma';
+
+const lazyReq = require('lazy-req')(require);
```
Comment (sindresorhus) [inline index.js]: This one can use the import syntax, and then just call the imported function.

File: index.js [hunk 5]
```diff
@@ -199,24 +218,22 @@ function lint() {
...
 
 	// workaround the errors array sometimes containing `null`
-	var errors = _.compact(linter.errors);
+	const errors = _.compact(linter.errors);
 
 	if (errors.length > 0) {
 		// aggregate same-line errors
-		var ret = [];
-		_.each(errors, function (el) {
+		let ret = [];
```
Comment (sindresorhus) [inline index.js]: Could be `const`.

File: load-config.js [hunk 6]
```diff
@@ -1,15 +1,16 @@
...
+import shjs from 'shelljs';
+import cli from 'jshint/src/cli';
+import userHome from 'user-home';
 
 // from JSHint //
 // Storage for memoized results from find file
 // Should prevent lots of directory traversal &
 // lookups when liniting an entire project
-var findFileResults = {};
+let findFileResults = {};
```
Comment (sindresorhus) [inline load-config.js]: Could be `const`.

**Comment by sindresorhus** [comment]
Awesome! :)

There's a few more places you could use `const`. `const` only prevent reassignment, not mutation, so it's fine.

**Comment by sindresorhus** [comment]
> I'm not quite sure how to get rid of the lazy requires, so I didn't :sweat_smile:

Yeah, not possible until the Module Loader API is specced. This is fine for now.

**Comment by nicolasbrugneaux** [comment]
I changed the few extra `let` :)

**Comment by sindresorhus** [comment]
Yay! :)

**Comment by arthurvr** [comment]
Thankkss!

**Pull request closed without merging**